src/ppx_deriving_qcheck/QCheck_generators.ml

```diff
@@ -5,92 +5,222 @@ open Ppxlib
 
 (** {2. Type} *)
 
-let ty = Ldot (Ldot (Lident "QCheck", "Gen"), "t")
+let ty = function
+  | `QCheck -> Ldot (Ldot (Lident "QCheck", "Gen"), "t")
+  | `QCheck2 -> Ldot (Ldot (Lident "QCheck2", "Gen"), "t")
```

Comment:
Would introducing a `VERSION` module and changing these  to use a `to_module` function improve this duplication?
```
module type VERSION = sig
  type t
  val version : t
  val to_module : t -> label
end

type version = [`QCheck | `QCheck2]

let to_module = function
  | `QCheck -> "QCheck"
  | `QCheck2 -> "QCheck2"
  
```

Reply:
Yes you're right, I believe we would never have more than 2 versions but that'd simplify the code a bit.

Reply:
Although, I don't see the real benefit of the module type `VERSION`.
Isn't:
```ocaml
type version = [`QCheck | `QCheck2]
                    
let to_module : version -> string = function
  | `QCheck -> "QCheck"
  | `QCheck2 -> "QCheck2"

let ty version = Ldot (Ldot (Lident (to_module version), "Gen"), "t")

...

module Make (Version : sig val version : version end) = struct
```
enough?

Reply:
It would be enough, I would go with that. 
Making it a module type might be slightly tidier if more than 2 versions were expected or there were other places that needed that type. In my hacking it didn't seem worthwhile.